File: backend/src/main/java/az/project/VocabularyBuilder/tictactoe/dto/GameReplayDto.java
package az.project.VocabularyBuilder.tictactoe.dto;

import az.project.VocabularyBuilder.tictactoe.domain.GameResult;
import az.project.VocabularyBuilder.tictactoe.domain.GameType;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record GameReplayDto(
        UUID sessionUuid,
        GameType gameType,
        GameResult result,
        String playerSide,
        LocalDateTime datePlayed,
        List<GameMoveDto> moves
) {
}

File: frontend/src/app/shared/models/gameReplayDto.ts
import { GameMoveDto } from "../../features/tic-tac-toe/models/gameMoveDto";
import { GameResult } from "../enums/gameResult";
import { GameType } from "../enums/gameType";

export interface GameReplayDto {
    sessionUuid: string;

    gameType: GameType;

    result: GameResult;

    playerSide: string;

    datePlayed: string;

    moves: GameMoveDto[];
}